pkg/api/serialization_test.go

```diff
@@ -144,6 +146,40 @@ func roundTripSame(t *testing.T, group testapi.TestGroup, item runtime.Object, e
 	}
 }
 
+func TestSetControllerConversion(t *testing.T) {
+	rs := &extensions.ReplicaSet{}
+	rc := &v1.ReplicationController{}
+
+	extGroup := testapi.Extensions
+	defaultGroup := testapi.Default
+
+	fuzzInternalObject(t, extGroup.InternalGroupVersion(), rs, rand.Int63())
+
+	// RS -> RC
+	data, err := runtime.Encode(extGroup.Codec(), rs)
+	if err != nil {
+		t.Fatalf("unexpected encoding error: %v", err)
+	}
+
+	codec := extGroup.Codec()
+	if err := versioning.EnableCrossGroupDecoding(codec, extGroup.GroupVersion().Group, defaultGroup.GroupVersion().Group); err != nil {
+		t.Fatalf("unexpected error while enabling cross-group decoding: %v", err)
+	}
+	if err := runtime.DecodeInto(codec, data, rc); err != nil {
```

Comment:
`data` is an `rs.v1beta1.extensions` and you have no conversion registered to decode it into an `rc.__internal`.  Wrote the test backwards maybe?